addons/account_update_tax_tags/wizard/account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,169 @@
+                SELECT aml.id AS aml_id, rep_tags.account_account_tag_id AS tag_id
+                  FROM account_move_line aml
+                  JOIN account_move move ON aml.move_id = move.id AND aml.date >= %(from_date)s
+                  JOIN account_tax_repartition_line rep_ln ON rep_ln.id = aml.tax_repartition_line_id
+             LEFT JOIN account_account_tag_account_tax_repartition_line_rel rep_tags ON rep_tags.account_tax_repartition_line_id = aml.tax_repartition_line_id
+                 WHERE move.id != ALL(%(excluded_move_ids)s)
+                       AND move.state != 'cancel' -- this is to avoid issue with legacy broken move
+                       AND (move.move_type != 'entry' OR %(accept_entry_type_move)s)
+                       AND rep_ln.id IN %(rep_ln_ids)s
+            ), base_and_tax_aml_tag_id AS (
+                SELECT barti.aml_id, barti.tag_id FROM base_aml_id_rep_tag_to_insert barti
+                 UNION
+                SELECT tarti.aml_id, tarti.tag_id FROM tax_aml_id_rep_tag_to_insert tarti
+            ), delete_statement AS (
+                -- ----------------------------------
+                -- 2) Delete previous tags from amls
+                -- ----------------------------------
+                DELETE FROM account_account_tag_account_move_line_rel aml_tags
+                      USING base_and_tax_aml_tag_id btat
+                      WHERE btat.aml_id = aml_tags.account_move_line_id
+                  RETURNING account_move_line_id AS aml_id
+            ), insert_statement AS (
+                -- --------------------------
+                -- 3) Insert new tag on amls
+                -- --------------------------
+                INSERT INTO account_account_tag_account_move_line_rel
+                     SELECT btat.aml_id, btat.tag_id
+                       FROM base_and_tax_aml_tag_id btat
+                      WHERE btat.tag_id NOTNULL
+                  RETURNING account_move_line_id AS aml_id
+            ), impacted_aml as (
+                -- ------------------------
+                -- 4) Return impacted amls
+                -- ------------------------
+                SELECT ds.aml_id AS aml_id
+                  FROM delete_statement ds
+                 UNION
+                SELECT ins.aml_id AS aml_id
+                  FROM insert_statement ins
+            )
+            SELECT ARRAY_AGG(impacted_aml.aml_id)
+              FROM impacted_aml
+        """, params={
+            'from_date': from_date,
+            'rep_ln_ids': tuple(repartition_lines_ids),
+            'excluded_move_ids': excluded_move_ids,
+            'accept_entry_type_move': accept_entry_type_move,
+        })
+        self.flush()
+        self.invalidate_cache()
+        # return the postgresql array of ids
+        return self.env.cr.fetchone()[0]
+
+    def update_amls_tax_tags(self):
+        self._modify_tag_to_aml_relation(self.company_id.id, self.from_date)
```

Comment:
Ideally, that would need to recompute the `tax_audit_string` but that might be very slow